_studio/mfx_lib/shared/src/libmfxsw_decode.cpp

```diff
@@ -417,25 +413,34 @@ mfxStatus MFXVideoDECODE_DecodeHeader(mfxSession session, mfxBitstream *bs, mfxV
 mfxStatus MFXVideoDECODE_Init(mfxSession session, mfxVideoParam *par)
 {
     mfxStatus mfxRes;
-    MFX_CHECK(par, MFX_ERR_NULL_PTR);
 
     MFX_AUTO_LTRACE_FUNC(MFX_TRACE_LEVEL_API);
     MFX_LTRACE_BUFFER(MFX_TRACE_LEVEL_API, par);
 
+    MFX_CHECK(session, MFX_ERR_INVALID_HANDLE);
+    MFX_CHECK(par, MFX_ERR_NULL_PTR);
+
+    try
+    {
         // check existence of component
-        if (!session->m_pDECODE.get())
+        if (!session->m_pDECODE)
         {
             // create a new instance
             session->m_pDECODE.reset(session->Create<VideoDECODE>(*par));
             MFX_CHECK(session->m_pDECODE.get(), MFX_ERR_INVALID_VIDEO_PARAM);
```

Comment:
.get() is unnecessary in such checks

Reply:
It is unnesessary in the honest if (smart_ptr), but our MFX_CHECK transforms to if (!!!(smart_ptr)) and compiler goes crazy.

Reply:
got it